frontend/e2e/$lang+/_public/apply/adult/adult-flow.spec.ts

```diff
@@ -0,0 +1,477 @@
+
+  await page.getByRole('group', { name: 'Federal benefits' }).getByRole('radio', { name: 'Yes, I have federal dental benefits' }).check();
+  await page.getByRole('group', { name: 'Provincial or territorial benefits' }).getByRole('radio', { name: 'Yes, I have provincial or territorial dental benefits' }).check();
+  await page.getByRole('button', { name: 'Continue' }).click();
+
+  // check for empty fields
+  await hasError(page, 'Select which federal program you are covered under');
+  await hasError(page, 'Select a province or territory');
+
+  await page.getByRole('group', { name: 'Federal benefits' }).getByRole('radio', { name: 'Correctional Service Canada Health Services' }).check();
+  await page.getByRole('group', { name: 'Provincial or territorial benefits' }).getByRole('combobox', { name: 'If yes, through which province or territory?' }).selectOption('Alberta');
+  await page.getByRole('button', { name: 'Continue' }).click();
+
+  await hasError(page, 'Select which provincial or territorial program you are covered under');
+  await page.getByRole('group', { name: 'Provincial or territorial benefits' }).getByRole('radio', { name: 'Alberta Adult Health Benefits' }).check();
+  await page.getByRole('button', { name: 'Continue' }).click();
+}
+
+test.describe('Adult flow', () => {
+  test.beforeEach('Navigate to online application', async ({ page }) => {
+    test.setTimeout(60000);
+    await page.goto('/en/apply');
+    await expect(page).toHaveURL(/\/en\/apply\/[a-f0-9-]+\/terms-and-conditions/);
+    await expect(page.getByRole('heading', { level: 1, name: 'Terms and conditions' })).toBeVisible();
+    await page.getByRole('button', { name: 'Agree and continue' }).click();
+  });
+
+  test('should complete application as adult applicant', async ({ page }) => {
+    await test.step('Type of application page', async () => {
+      await expect(page).toHaveURL(/\/en\/apply\/[a-f0-9-]+\/type-application/);
+      await expect(page.getByRole('heading', { level: 1, name: 'Type of application' })).toBeVisible();
+      await page.getByRole('button', { name: 'Continue' }).click();
+
+      //check for empty fields
+      await hasError(page, 'Select who this application is for');
+
+      await page.getByRole('radio', { name: 'I am applying for myself', exact: true }).check();
+      await page.getByRole('button', { name: 'Continue' }).click();
+    });
+
+    await test.step('Tax filing page', async () => {
+      await expect(page).toHaveURL(/\/en\/apply\/[a-f0-9-]+\/adult\/tax-filing/);
+      await expect(page.getByRole('heading', { level: 1, name: 'Tax filing' })).toBeVisible();
+      await page.getByRole('button', { name: 'Continue' }).click();
+
+      //check for empty fields
+      await hasError(page, 'Select whether or not you have filed your taxes');
+
+      await page.getByRole('radio', { name: 'No', exact: true }).check();
+      await page.getByRole('button', { name: 'Continue' }).click();
+    });
```

Comment:
This "Tax filing page" step appears to be repeated - can it be extracted into a reusable function as well?